src/components/List.js

```diff
@@ -1,12 +1,54 @@
 import React from 'react';
 import useFirestore from '../hooks/useFirestore';
+import { db } from '../lib/firebase';
 
 const List = ({ token }) => {
   const { docs } = useFirestore(token);
+
+  const checkPurchasedDate = (purchasedDate) => {
+    if (purchasedDate === null) {
+      return false;
+    }
+    const today = new Date();
+    let daySincePurchased = purchasedDate.toDate();
+
+    daySincePurchased.setDate(daySincePurchased.getDate() + 1); // Add one day to the last purchased date
+
+    // If today > daySincePurchased at least 24 hours have passed, return false to uncheck box
+    if (
+      Date.parse(today.toString()) >= Date.parse(daySincePurchased.toString())
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleCheckbox = async (event) => {
+    const currentDate = new Date();
+    const queryCollection = await db.collection(token).doc(event.target.id);
+    queryCollection.update({
+      lastPurchased: currentDate,
+    });
+  };
   return (
     <div>
       <h1>List</h1>
-      <ul>{docs && docs.map((doc) => <li key={doc.id}>{doc.itemName}</li>)}</ul>
+      <ul style={{ listStyleType: 'none' }}>
+        {docs &&
+          docs.map((doc) => (
+            <li key={doc.id}>
+              <input
+                type="checkbox"
+                name={doc.itemName}
+                id={doc.id}
+                onChange={handleCheckbox}
```

Comment:
Right now, if the user clicks on an already checked checkbox it will stay locked in the checked state. 
What if the user accidentally clicked the wrong one by mistake? Being able to clear the checkbox might be useful.

This wasn't explicitly called out in the acceptance criteria, so don't feel like you need to make this change. Just an idea worth thinking about!

Reply:
Thanks @chrisdopuch. We noticed that issue as well and that was brought up during office hours today. It sounds like something that will be handled down the road

Reply:
Sounds good to me, glad you are thinking about it too 🙌